From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: William Blake Galbreath <Blake.Galbreath@GMail.com>
Date: Sat, 3 Oct 2020 21:39:16 -0500
Subject: [PATCH] Entity#isTicking


diff --git a/src/main/java/net/minecraft/server/Entity.java b/src/main/java/net/minecraft/server/Entity.java
index 239945cf657bfb94dbd935d64cfdef082e24fbad..a2651528ec2952cfe8049f185c21d3577fa1e1f5 100644
--- a/src/main/java/net/minecraft/server/Entity.java
+++ b/src/main/java/net/minecraft/server/Entity.java
@@ -3386,5 +3386,9 @@ public abstract class Entity implements INamableTileEntity, ICommandListener, Ke
     public static int nextEntityId() {
         return entityCount.incrementAndGet();
     }
+
+    public boolean isTicking() {
+        return ((ChunkProviderServer) world.getChunkProvider()).isInEntityTickingChunk(this);
+    }
     // Paper end
 }
diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java
index f1243b308e44d7c08e007d8ce080c2fc6248753c..4ec0e9220d2fa976dc6dfecd3eedb5b78a0ef8f4 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java
@@ -1143,5 +1143,9 @@ public abstract class CraftEntity implements org.bukkit.entity.Entity {
     public boolean isInLava() {
         return getHandle().isInLava();
     }
+
+    public boolean isTicking() {
+        return getHandle().isTicking();
+    }
     // Paper end
 }
